services/device-registry-base/src/test/java/org/eclipse/hono/service/management/device/AbstractDeviceManagementSearchDevicesTest.java

```diff
@@ -0,0 +1,245 @@
+package org.eclipse.hono.service.management.device;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.hono.deviceregistry.util.DeviceRegistryUtils;
+import org.junit.jupiter.api.Test;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Future;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * As suite of tests for verifying implementations of the Device management's 
+ * search devices operation.
+ * <p>
+ * Concrete subclasses need to provide the service implementations under test
+ * by means of implementing the {@link #getDeviceManagementService()} method.
+ * Also the subclasses should clean up any fixture in the database that has
+ * been created by individual test cases.
+ */
+public abstract class AbstractDeviceManagementSearchDevicesTest {
+
+    /**
+     * Gets device management service being tested.
+     *
+     * @return The device management service
+     */
+    public abstract DeviceManagementService getDeviceManagementService();
+
+    /**
+     * Verifies that a request to search devices fails with a {@value HttpURLConnection#HTTP_NOT_FOUND}
+     * when no matching devices are found.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testSearchDevicesWhenNoDevicesAreFound(final VertxTestContext ctx) {
+        final String deviceId = DeviceRegistryUtils.getUniqueIdentifier();
+        final String tenantId = DeviceRegistryUtils.getUniqueIdentifier();
+        final int pageSize = 10;
+        final int pageOffset = 0;
+        final Filter filter = new Filter("/enabled", false);
+
+        createDevices(tenantId, Map.of(deviceId, new Device()))
+                .compose(ok -> getDeviceManagementService()
+                        .searchDevices(tenantId, pageSize, pageOffset, List.of(filter), List.of(), NoopSpan.INSTANCE))
+                .onComplete(ctx.succeeding(s -> {
+                    ctx.verify(() -> {
+                        assertThat(s.isError());
+                        assertThat(s.getStatus()).isEqualTo(HttpURLConnection.HTTP_NOT_FOUND);
+                    });
+                    ctx.completeNow();
+                }));
+    }
+
+    /**
+     * Verifies that a request to search devices with a valid filter succeeds and matching devices are found.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testSearchDevicesWithAFilterSucceeds(final VertxTestContext ctx) {
+        final String tenantId = DeviceRegistryUtils.getUniqueIdentifier();
+        final int pageSize = 10;
+        final int pageOffset = 0;
+        final Filter filter = new Filter("/enabled", true);
+
+        createDevices(tenantId, Map.of(
+                "testDevice1", new Device().setEnabled(true),
+                "testDevice2", new Device().setEnabled(false)))
```

Comment:
IMHO we should also use random device IDs in order to minimize the potential for conflicts ...